wagtail_localize_rws_languagecloud/templatetags/wagtaillocalizerwslanguagecloud_tags.py

```diff
@@ -0,0 +1,38 @@
+from django import template
+
+from ..models import LanguageCloudFile
+
+
+register = template.Library()
+
+
+@register.simple_tag(takes_context=True)
+def language_cloud_statuses(context):
+    page = context.get("parent_page") or context.get("page")
+
+    if not page:
+        return []
+
+    locale_ids = page.get_translations(inclusive=True).values_list("locale", flat=True)
+
+    locale_id_combined_status_map = {}
+
+    for locale_id in locale_ids:
+        # Find the LanguageCloudFile for each locale
+        try:
+            lc_file = (
+                LanguageCloudFile.objects.filter(
+                    translation__source__object_id=page.translation_key,
+                    translation__target_locale=locale_id,
+                )
+                .select_related("project", "translation")
+                .prefetch_related("project__languagecloudfile_set")
+                .latest("project__source_last_updated_at")
+            )
+        except LanguageCloudFile.DoesNotExist:
+            continue
+
+        if lc_file:
+            locale_id_combined_status_map[locale_id] = lc_file.combined_status
```

Comment:
As we discussed in slack, we should be able to make this more efficient by building up the queryset objects in the loop then `union()`-ing them together before we evaluate so we only do one round-trip to the DB.

Reply:
I will change this!

Reply:
I tried doing this but unfortunately it didn't work. Here's what I tried:

```python
querysets = [
    LanguageCloudFile.objects.filter(
        translation__source__object_id=translation_key,
        translation__target_locale=locale_id,
    )
    .select_related("project", "translation")
    .prefetch_related("project__languagecloudfile_set")
    .order_by("-project__source_last_updated_at")[:1]
    for locale_id in locale_ids
]

lc_files = reduce(lambda qs1, qs2: qs1.union(qs2), querysets)
```

This error message is returned when I try to evaluate `lc_files`.

`*** django.db.utils.DatabaseError: LIMIT/OFFSET not allowed in subqueries of compound statements.`

Ultimately, we want to get the **latest** LanguageCloud file according to `project__source_last_updated_at`. Using `.get`, `.latest`, or `.first` doesn't work because those return single instances and not `QuerySet` objects that we can `.union`. Slicing looked promising until I realized most databases don't support that operation when using combined queries.

Reply:
Bah. Sorry for sending you on a goose chase with that one! Lets not sink any more time into this